File: DoohClick.API/DoohClick.Model/Application/Inv/Screen/ScreenModel.cs


namespace DoohClick.Model.Application.Inv.Screen
{
    public record MvScreenFilterOptions
    {
        public required int TenantId { get; set; }
        public bool IsActive { get; set; } = true;
        public List<string>? CountryCodeList { get; set; }
        public List<string>? CityList { get; set; }
        public List<string>? OrientationList { get; set; }
        public List<string>? ResolutionList { get; set; }
    }
    public record MvScreen
    {
        public int? Id { get; set; }
        public Guid? Uuid { get; set; }
        public int? TenantId { get; set; }
        public string? TenantName { get; set; }
        public required string Name { get; set; }
        public string? NormalizedName { get; set; }
        public required string ScreenCode { get; set; }
        public string? Description { get; set; }
        public required string DefaultResolution { get; set; }
        public required string Orientation { get; set; }
        public required string Location { get; set; }
        public string? AddressLine { get; set; }
        public List<string>? Tag { get; set; }
        public required string CountryCode { get; set; }
        public required string City { get; set; }
        public required string Timezone { get; set; }
        public bool IsActive { get; set; }
        public decimal RatePerHour { get; set; }
        public required string Currency { get; set; }
        public List<MvScreenOperatingHour>? OperatingHour { get; set; } = [];
        public List<MvScreenSupportedMedia>? SupportedMedia { get; set; } = [];
        public int? CreatedBy { get; set; }
        public string? Creator { get; set; }
        public int? UpdatedBy { get; set; }
        public string? Modifier { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }

    public record MvScreenOperatingHour
    {
        public int? Id { get; set; }
        public required string DayOfWeek { get; set; }
        public required string OpenTime { get; set; }
        public required string CloseTime { get; set; }
        public required string AudienceSource { get; set; }
        public int? EstimatedImpression { get; set; }
    }

    public record MvScreenSupportedMedia
    {
        public int? Id { get; set; }
        public required string MediaType { get; set; }
    }

    public record MvScreenDelParam
    {
        public required Guid Uuid { get; set; }
        public required int TenantId { get; set; }
        public required int DeletedBy { get; set; }
    }
}

File: DoohClick.Web/src/app/inv/screen/model/screen.model.ts
export interface MvScreen {
  id?: number;
  uuid?: string;
  tenantId?: number;
  tenantName?: string;
  name: string;
  normalizedName?: string;
  screenCode: string;
  description?: string;
  defaultResolution: string;
  orientation: string;
  location: string;
  addressLine?: string;
  tag?: string[];
  countryCode: string;
  city: string;
  timezone: string;
  isActive: boolean;
  ratePerHour: number;
  currency: string;
  operatingHour?: MvScreenOperatingHour[];
  supportedMedia?: MvScreenSupportedMedia[];
  createdBy?: number;
  creator?: string;
  updatedBy?: number;
  modifier?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface MvScreenOperatingHour {
  id?: number;
  dayOfWeek: string;
  openTime: string;
  closeTime: string;
  audienceSource: string;
  estimatedImpression?: number;
  deletedBy?: number;
}

export interface MvScreenSupportedMedia {
  id?: number;
  mediaType: string;
  deletedBy?: number;
}

export interface MvScreenDelParam {
  uuid: string;
  tenantId?: number;
  deletedBy?: number;
}

export interface MvScreenFilterOptions {
  tenantId: number;
  isActive?: boolean;
  countryCodeList?: string[];
  cityList?: string[];
  orientationList?: string[];
  resolutionList?: string[];
}

export interface MvScreenDdl {
  id?: number;
  name: string;
  screenCode: string;
  defaultResolution: string;
  orientation: string;
  tag?: string[];
  countryCode: string;
  city: string;
  timezone: string;
  operatingHour: MvScreenOperatingHour[];
  supportedMedia: MvScreenSupportedMedia[];
}
